IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -96,4 +96,130 @@ + (nullable NSDictionary *)getRegisteredDeviceMetadataInformation:(nullable id<M
     return nil;
 }
 
++ (nullable MSIDWPJKeyPairWithCert *)findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:(nonnull NSDictionary *)queryAttributes
```

Comment:
This code was moved almost unchanged from IdentityCore/src/workplacejoin/Mac/MSIDWorkPlaceJoinUtil.m.
Changes:
1. Accept custom private key and cert attributes
2. Set kSecReturnRef for cert query (for some reason it wasn't set on macOS, and this was failing on iOS -TODO add a test case to validate)
3. Update to iOS style APIs - SecCertificateCopyKey/SecCertificateCopyNormalizedIssuerSequence

Reply:
Issuer is used in MSAL Obj-C and MSAL CPP to do issuer matching against the challenge. I'm not sure if using a displayable version with Apple's own way to convert it would break the matching code (it has been there forever). If you're confident that it won't break existing logic, I can switch to SecCertificateCopyValues, otherwise would like to keep it as is.

Reply:
I think the issuer of a x509 certificate is represented as a ASN.1 sequence of key,value pairs. When this sequence is decoded as a string it will look something like : 'Issuer: C=BE, O=GlobalSign nv-sa, OU=Root CA, CN=GlobalSign Root CA' . 

For PKeyAuth we use regex to parse out 'OU=<organizational unit value>' from its string decoding done at line 90 of [IdentityCore/src/cache/crypto/MSIDWPJKeyPairWithCert.m](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1083/files#diff-08e04ebe5d940e3a795366b8ab430f6de5203f3cc5a0242363d0e496d789e5fe) and compare it with issuer in challenge. This is done in MSIDPKeyAuthHelper class for workplacejoin as [MSIDPkeyAuthHelper getOrgUnitFromIssuer:[info certificateIssuer]];

If we use [SecCertificateCopyValues](https://developer.apple.com/documentation/security/1396051-seccertificatecopyvalues?language=objc) and supply it with OID for issuer organizational unit : [kSecOIDOrganizationalUnitName](https://developer.apple.com/documentation/security/ksecoidorganizationalunitname?language=objc), some of the complicated logic of string regex parsing can be avoided that is done in MSIDPKeyAuthHelper class.

I think it will be good to get rid of regex matching and parsing the decoded issuer string done in MSIDPKeyAuthHelper.

Reply:
I didn't see it before but SecCertificateCopyValues is available only on macOS and not iOS.